token/hmac/hmacsha.go

```diff
@@ -91,34 +88,36 @@ func (c *HMACStrategy) Generate(ctx context.Context) (string, string, error) {
 func (c *HMACStrategy) Validate(ctx context.Context, token string) (err error) {
 	var keys [][]byte
 
-	secrets, err := c.Config.GetGlobalSecret(ctx)
+	globalSecret, err := c.Config.GetGlobalSecret(ctx)
 	if err != nil {
 		return err
 	}
 
+	if len(globalSecret) > 0 {
+		keys = append(keys, globalSecret)
+	}
+
 	rotatedSecrets, err := c.Config.GetRotatedGlobalSecrets(ctx)
 	if err != nil {
 		return err
 	}
 
-	if len(secrets) > 0 {
-		keys = append(keys, secrets)
+	keys = append(keys, rotatedSecrets...)
+
+	if len(keys) == 0 {
```

Comment:
Took me quite some time to figure out that the error check after the loop was just this...